autopush/endpoint.py

```diff
@@ -877,11 +918,20 @@ def _success(self, result):
     #                    Utility Methods
     #############################################################
     def _validate_auth(self, uaid):
-        """Validates the Authorization header in a request"""
-        secret = self.ap_settings.crypto_key
-        hashed = self.request.headers.get("Authorization", "").strip()
-        key = generate_hash(secret, uaid)
-        return validate_hash(key, self.request.body, hashed)
+        """Validates the Authorization header in a request
+
+        Validate the given request using HAWK.
+        """
+        secret = generate_hash(self.ap_settings.crypto_key, uaid)
+
+        fReq = prequests.Request(
+            self.request.method,
+            "%s://%s%s" % (self.request.protocol, self.request.host,
+                           self.request.uri),
+            headers=self.request.headers,
+            data=self.request.body).prepare()
+        return hawkauthlib.check_signature(fReq,
+                                           secret)
```

Comment:
Fast enough that it'd be worse to kick it to a thread. So I guess well call them "not blocking enoug"? ;)